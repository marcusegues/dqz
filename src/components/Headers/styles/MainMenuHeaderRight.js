// @flow
// $FlowFixMe
import { StyleSheet } from 'react-native';

export const mainMenuHeaderRightStyle = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
});
