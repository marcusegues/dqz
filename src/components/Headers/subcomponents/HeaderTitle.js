import React from 'react';

import { Text } from 'react-native';

const HeaderTitle = ({ text }) => (
  <Text
    style={{
      fontSize: 18,
      color: '#1A1A1A',
      fontFamily: 'roboto_regular',
    }}
  >
    {text}
  </Text>
);

export default HeaderTitle;
