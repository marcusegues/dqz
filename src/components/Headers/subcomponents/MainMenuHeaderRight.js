import React from 'react';
import { View } from 'react-native';
import ReceiptNotificationBadge from './ReceiptNotificationBadge';
import OptionsGearIcon from './OptionsGearIcon';
import ownStyles from '../styles/MainMenuHeaderRight';

const MainMenuHeaderRight = ({ navigation }) => (
  <View style={ownStyles.container}>
    <ReceiptNotificationBadge />
    <OptionsGearIcon />
  </View>
);

export default MainMenuHeaderRight;
