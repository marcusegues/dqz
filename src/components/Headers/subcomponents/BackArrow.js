import React from 'react';
import { Ionicons } from '@expo/vector-icons';
import Touchable from 'react-native-platform-touchable';
import { MAIN_RED } from '../../../styles/colors';

const BackArrow = ({ onPress }) => (
  <Touchable onPress={onPress}>
    <Ionicons name="md-arrow-back" size={32} color={MAIN_RED} />
  </Touchable>
);

export default BackArrow;
