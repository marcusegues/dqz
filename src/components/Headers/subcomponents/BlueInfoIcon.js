// @flow
import React from 'react';
// $FlowFixMe
import Touchable from 'react-native-platform-touchable';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import { moderateScale } from '../../../styles/Scaling';

export const BlueInfoIcon = () => (
  <Touchable onPress={() => {}}>
    <MaterialCommunityIcons
      name="information"
      size={moderateScale(30)}
      color="#006699"
    />
  </Touchable>
);
