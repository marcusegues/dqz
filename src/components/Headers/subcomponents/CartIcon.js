// @flow
import React from 'react';
// $FlowFixMe
import Touchable from 'react-native-platform-touchable';
import { MaterialIcons } from '@expo/vector-icons';
import { moderateScale } from '../../../styles/Scaling';
import type { Navigation } from '../../../types/generalTypes';
import { receiptNotificationBadgeStyle } from '../styles/ReceiptNotificationBadge';
import { MAIN_RED } from '../../../styles/colors';

type CartIconProps = {
  navigation: Navigation,
};

export const CartIcon = ({ navigation }: CartIconProps) => (
  <Touchable
    onPress={() =>
      navigation.dispatch({ type: 'NAVIGATE', screen: 'QuestionAnswer' })
    }
  >
    <MaterialIcons
      name="shopping-cart"
      size={moderateScale(30)}
      color={MAIN_RED}
      style={receiptNotificationBadgeStyle.receiptIcon}
    />
  </Touchable>
);
