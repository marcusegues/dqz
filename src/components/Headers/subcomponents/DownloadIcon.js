// @flow
import React from 'react';
// $FlowFixMe
import Touchable from 'react-native-platform-touchable';
import { MaterialIcons } from '@expo/vector-icons';
import { MAIN_RED } from '../../../styles/colors';
import { moderateScale } from '../../../styles/Scaling';
import type { Navigation } from '../../../types/generalTypes';

type DownloadIconProps = {
  navigation: Navigation,
};

export const DownloadIcon = ({ navigation }: DownloadIconProps) => (
  <Touchable
    onPress={() => {
      navigation.goBack();
    }}
  >
    <MaterialIcons name="get-app" size={moderateScale(28)} color={MAIN_RED} />
  </Touchable>
);
