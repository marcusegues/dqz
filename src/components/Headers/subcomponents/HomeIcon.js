// @flow
import React from 'react';
// $FlowFixMe
import Touchable from 'react-native-platform-touchable';
import Entypo from '@expo/vector-icons/Entypo';
import { moderateScale } from '../../../styles/Scaling';
import type { Navigation } from '../../../types/generalTypes';
import { GREY } from '../../../styles/colors';

type HomeIconProps = {
  navigation: Navigation,
};

export const HomeIcon = ({ navigation }: HomeIconProps) => (
  <Touchable
    onPress={() =>
      navigation.dispatch({ type: 'NAVIGATE', screen: 'MainMenu' })
    }
  >
    <Entypo name="home" size={moderateScale(27)} color={GREY} />
  </Touchable>
);
