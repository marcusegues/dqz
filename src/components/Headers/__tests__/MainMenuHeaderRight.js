// @flow
import React from 'react';
import { shallow } from 'enzyme';
import { MainMenuHeaderRight } from '../subcomponents/MainMenuHeaderRight';

describe('MainMenuHeaderRight Icon TestSuite', () => {
  test('renders as expected', () => {
    const clickFn = jest.fn();
    const wrapper = shallow(
      // $FlowFixMe
      <MainMenuHeaderRight navigation={clickFn} language="de" />
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
