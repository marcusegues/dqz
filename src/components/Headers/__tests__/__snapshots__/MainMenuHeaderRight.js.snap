// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenuHeaderRight Icon TestSuite  as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Connect(ReceiptNotificationBadgeInner)
      navigation={[Function]}
    />
    <LanguageSelect
      language="de"
      navigation={[Function]}
    />
  </View>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(ReceiptNotificationBadgeInner)
          navigation={[Function]}
        />,
        <LanguageSelect
          language="de"
          navigation={[Function]}
        />,
      ],
      "style": Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "navigation": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "language": "de",
          "navigation": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(ReceiptNotificationBadgeInner)
            navigation={[Function]}
          />,
          <LanguageSelect
            language="de"
            navigation={[Function]}
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "navigation": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "language": "de",
            "navigation": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MainMenuHeaderRight Icon TestSuite renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Connect(ReceiptNotificationBadgeInner)
      navigation={[Function]}
    />
    <LanguageSelect
      language="de"
      navigation={[Function]}
    />
  </View>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(ReceiptNotificationBadgeInner)
          navigation={[Function]}
        />,
        <LanguageSelect
          language="de"
          navigation={[Function]}
        />,
      ],
      "style": Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "navigation": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "language": "de",
          "navigation": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(ReceiptNotificationBadgeInner)
            navigation={[Function]}
          />,
          <LanguageSelect
            language="de"
            navigation={[Function]}
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "navigation": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "language": "de",
            "navigation": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
